model/pdata/metrics.go

```diff
@@ -350,3 +351,8 @@ func (ms Exemplar) Type() MetricValueType {
 	}
 	return MetricValueTypeNone
 }
+
+// NewOptionalDouble creates an data.OptionalDouble from a float64.
+func NewOptionalDouble(val float64) *data.OptionalDouble {
+	return data.NewOptionalDouble(val)
+}
```

Comment:
1. "*data.OptionalDouble" is internal so you need to have something like "type OptionalDouble = data.OptionalDouble" and return that at least.
2. I think instead of changing `HistogramDataPoint.Sum(OptionalDouble)` we can have the same `HistogramDataPoint.Sum() -> float64` and have a `HistogramDataPoint.HasSum() -> bool`?